src/lib/fauna-client.mjs

```diff
@@ -179,3 +179,70 @@ export const formatQuerySummary = (summary) => {
     return summary;
   }
 };
+
+/**
+ * Selects a subset of query info fields from a v10 query response.
+ * @param {import("fauna").QueryInfo} response - The query response
+ * @param {string[]} include - The query info fields to include
+ * @returns {object} An object with the selected query info fields
+ */
+const pickAndCamelCaseQueryInfo = (response, include) => {
+  const queryInfo = {};
+
+  if (include.includes("txnTs") && response.txn_ts)
+    queryInfo.txnTs = response.txn_ts;
+  if (include.includes("schemaVersion") && response.schema_version)
+    queryInfo.schemaVersion = response.schema_version.toString();
+  if (include.includes("summary") && response.summary)
+    queryInfo.summary = response.summary;
+  if (include.includes("queryTags") && response.query_tags)
+    queryInfo.queryTags = response.query_tags;
+  if (include.includes("stats") && response.stats)
+    queryInfo.stats = response.stats;
+
+  return queryInfo;
+};
+
+/**
+ *
+ * @param {object} response - The v4 or v10 query response with query info
+ * @param {object} opts
+ * @param {string} opts.apiVersion - The API version
+ * @param {boolean} opts.color - Whether to colorize the error
+ * @param {string[]} opts.include - The query info fields to include
+ * @returns
+ */
+export const formatQueryInfo = (response, { apiVersion, color, include }) => {
+  if (apiVersion === "4" && include.includes("stats")) {
+    /** @type {import("faunadb").MetricsResponse} */
+    const metricsResponse = response;
+    const colorized = colorize(
+      { metrics: metricsResponse.metrics },
+      { color, format: Format.YAML },
+    );
+
+    return `${colorized}\n`;
+  } else if (apiVersion === "10") {
+    const queryInfoToDisplay = pickAndCamelCaseQueryInfo(response, include);
+
+    if (Object.keys(queryInfoToDisplay).length === 0) return "";
+
+    const SUMMARY_IN_QUERY_INFO_FQL_REGEX = /^(\s\s\s\s\|)|(\d\s\|)/;
+    const colorized = colorize(queryInfoToDisplay, {
+      color,
+      format: Format.YAML,
+    })
+      .split("\n")
+      .map((line) => {
+        if (!line.match(SUMMARY_IN_QUERY_INFO_FQL_REGEX)) {
```

Comment:
how about using the summary property & has-ansi to decide if we should colorize each block of queryInfoToDisplay? hopefully it'll be less brittle to summary additions/changes and any changes we make to our rendering of it.

e.g.:
```
let infoString = ""
for (const key of Object.keys(queryInfoToDisplay)) {
  key === 'summary'
    ? infoString += colorize(queryInfoToDisplay, { color, format: Format.FQL })
    : infoString += colorize(queryInfoToDisplay, { color, format: Format.YAML })
}
```

Reply:
It's a good call out on how brittle this bit is. 🤔 

We could split each key out more like
```js
key === 'summary'
  ? infoString += colorize({ [key]: queryInfoToDisplay[key], { color, format: Format.YAML })
  : infoString += formatQuerySummaryWithHeaderBits(queryInfoToDisplay.summary)
```

But check out the existing `formatQuerySummary` function. The first lines should NOT be formatted as FQL. Only the FQL hints should be, making formatting the summary quite an odd ball.

Out of scope for this PR is piping the included fields into ServiceError handling. Currently, rendering summary for errors is different than for successful queries. I think that's okay as is, but I don't know what it should be like for errors if we start rendering more info like stats.

I'd like to circle back to this once this is out and once Eric's error handling updates are in too. There may be some more common ground for rendering summary in a more reusable way. Even if not, following up to make sure summary rendering doesn't accidentally clobber rendering other output is smart. I'll file something to make sure we don't miss it, but I don't want to overdo it now while other things are in flux. Fair?